code/game/jobs/job_exp.dm

```diff
@@ -0,0 +1,207 @@
+#define EXP_TYPE_EXEMPT			"Exempt"
+
+// Globals, Defines, Etc
+
+
+
+
+// Admin Verbs
+
+
+/client/proc/cmd_admin_check_player_exp()	//Allows admins to determine who the newer players are.
+	set category = "Admin"
+	set name = "Check Player Playtime"
+	if(!check_rights(R_ADMIN))
+		return
+	var/msg = "<html><head><title>Playtime Report</title></head><body>Playtime:<BR><UL>"
+	for(var/client/C in clients)
+		var/list/play_records = params2list(C.prefs.exp)
+		var/gen_exp_time = text2num(play_records[EXP_TYPE_CREW])
+		gen_exp_time = num2text(round(gen_exp_time / 60)) + "h"
+		msg += "<LI> - [key_name_admin(C.mob)]: <A href='?_src_=holder;getplaytimewindow=\ref[C.mob]'>" + C.get_exp_general() + "</a></LI>"
+	msg += "</UL></BODY></HTML>"
+	src << browse(msg, "window=Player_playtime_check")
+
+
+/client/proc/cmd_show_exp_panel(var/client/C in clients)
+	if(!check_rights(R_ADMIN))
+		return
+	var/body = "<html><head><title>Playtime for [C.key]</title></head><BODY><BR>Playtime:"
+	body += C.get_exp_report()
+	body += "</BODY></HTML>"
+	src << browse(body, "window=playerplaytime;size=550x615")
+
+
+// Procs
+
+
+/datum/job/proc/available_in_playtime(client/C)
+	if(!C)
+		return 0
+	if(!exp_requirements || !exp_type)
+		return 0
+	if(!job_is_xp_locked(src.title))
+		return 0
+	if(check_rights(R_ADMIN, 0, C.mob))
+		return 1
+	var/list/play_records = params2list(C.prefs.exp)
+	var/isexempt = text2num(play_records[EXP_TYPE_EXEMPT])
+	if(isexempt)
+		return 0
+	var/my_exp = text2num(play_records[exp_type])
+	var/job_requirement = text2num(exp_requirements)
+	if(config.use_exp_restrictions_heads_hours && ((title in command_positions) || title == "AI"))
+		job_requirement = config.use_exp_restrictions_heads_hours * 60
+	if(my_exp >= job_requirement)
+		return 0
+	else
+		return (job_requirement - my_exp)
+
+
+
+/proc/job_is_xp_locked(jobtitle)
+	if(!config.use_exp_restrictions_heads && ((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	if(!config.use_exp_restrictions_other && !((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	return 1
+
+
+/mob/proc/get_exp_report()
+	if(client)
+		return client.get_exp_report()
+	else
+		return "[src] has no client."
+
+/client/proc/get_exp_report()
+	if(!config.use_exp_tracking)
+		return "Tracking is disabled in the server configuration file."
+
+	var/list/play_records = params2list(prefs.exp)
+	if(!play_records.len)
+		return "[key] has no records."
+
+	var/return_text = "<UL>"
+
+	var/list/exp_data = list()
+	for(var/cat in exp_jobsmap)
+		if(text2num(play_records[cat]))
+			exp_data[cat] = text2num(play_records[cat])
+		else
+			exp_data[cat] = 0
+
+	for(var/dep in exp_data)
+		if(exp_data[dep] > 0)
+			if(dep == EXP_TYPE_EXEMPT)
+				return_text += "<LI>Exempt (all jobs auto-unlocked)</LI>"
+			else if(exp_data[EXP_TYPE_CREW] > 0)
+				return_text += "<LI>[dep] [get_exp_format(exp_data[dep])] (" + num2text(round(exp_data[dep]/exp_data[EXP_TYPE_CREW]*100)) + "%)</LI>"
```

Comment:
- You get crew playtime for playing any living mob. Even antags like nukeops. So you can't have antag exp but no crew exp.
- You can in theory ghost a lot and not have any crew playtime, but I'm changing this to it won't stop exp showing. Minor case though.
- +ing together strings is fixed.
- Having a separate 'time on server' category is pointless, EXP_TYPE_CREW + EXP_TYPE_GHOST together already equal this, and I'm already getting flak for complexity - I don't want to add more.

You get general playtime for *any* non-dead role. Including antags. So its not possible for someone to have antag exp but not crew exp. If you never actually join a round, but ghost a lot,